### ISSUE TITLE
FlightUnavailableError

### ISSUE THREAD

**Issue by Anaisdg** [InfluxCommunity/influxdb3-python#102]
### Specifications

* Client Version: v3
* InfluxDB Version: Cloud 3

This is from this community question: 
https://community.influxdata.com/t/python-influxdb-client-3-query-error-tutorial-code/34129/8

### Code sample to reproduce problem

```python
import os, time
from influxdb_client_3 import InfluxDBClient3, Point

host = “https://eu-central-1-1.aws.cloud2.influxdata.com/”
client = InfluxDBClient3(host=host, token=my_token, org=my_organisation)

data = {
“point1”: {
“location”: “Klamath”,
“species”: “bees”,
“count”: 23,
},
“point2”: {
“location”: “Portland”,
“species”: “ants”,
“count”: 30,
},
“point3”: {
“location”: “Klamath”,
“species”: “bees”,
“count”: 28,
},
“point4”: {
“location”: “Portland”,
“species”: “ants”,
“count”: 32,
},
“point5”: {
“location”: “Klamath”,
“species”: “bees”,
“count”: 29,
},
“point6”: {
“location”: “Portland”,
“species”: “ants”,
“count”: 40,
},
}

for key in data:
point = (
Point(“census”)
.tag(“location”, data[key][“location”])
.field(data[key][“species”], data[key][“count”])
)
client.write(database=my_database, record=point)
time.sleep(1) # separate points by 1 second

print(“Complete. Return to the InfluxDB UI.”)

query = “”“SELECT *
FROM ‘census’
“””

table = client.query(query=query, database=my_database, language=‘sql’)
df = table.to_pandas().sort_values(by=“time”)
print(df)
```

### Expected behavior

Return results. Thank you! 

### Actual behavior

{
“name”: “FlightUnavailableError”,
“message”: “Flight returned unavailable error, with message: empty address list: . gRPC client debug context: UNKNOWN:empty address list: {grpc_status:14, created_time:"2024-05-05T13:09:44.314680034+00:00"}. Client context: IOError: Server never sent a data message. Detail: Internal”,
“stack”: "---------------------------------------------------------------------------
FlightUnavailableError Traceback (most recent call last)
Cell In[3], line 6

**Comment by powersj** [comment]
From the link to the forums it looks like the user is running on Windows. The only other time I've seen this error is when the SSL certs are not installed correctly. Usually something like the following errors will appear as well:

```
E0727 14:32:06.973000000 22488 ssl_utils.cc:608] load_file: UNKNOWN:Failed to load file {created_time:"2023-07-27T21:32:06.97328587+00:00", filename:"/usr/share/grpc/roots.pem", children:[UNKNOWN:No such file or directory {created_time:"2023-07-27T21:32:06.973276213+00:00", errno:2, os_error:"No such file or directory", syscall:"fopen"}]}

E0727 14:32:06.974000000 22488 ssl_security_connector.cc:430] Could not get default pem root certs…..
```

There are steps on to do this in the README: https://github.com/InfluxCommunity/influxdb3-python/blob/70ceac05026a143cb8af953c306111149f89c06a/README.md?plain=1#L166-L200

**Comment by colin-jensen** [comment]
I ran into this problem on a mac as well. Is it because the cloud 3 python client is incompatible with cloud 2 accounts?

**Comment by powersj** [comment]
> Is it because the cloud 3 python client is incompatible with cloud 2 accounts?

Yes - the query component in the v3 python client is based on FlightSQL/gRPC, which is not used in v2. v2 users should use the v2 python client: https://github.com/influxdata/influxdb-client-python

**Comment by colin-jensen** [comment]
Gotcha. The tutorial on the website (Load Data -> Clients -> Python) directs me to use the v3 python client. Maybe v2 users could be directed to a different tutorial? As a new user, it took me while to figure out which version I was using and which client to use.

**Comment by FangqiZhu** [comment]
> From the link to the forums it looks like the user is running on Windows. The only other time I've seen this error is when the SSL certs are not installed correctly. Usually something like the following errors will appear as well:
> 
> ```
> E0727 14:32:06.973000000 22488 ssl_utils.cc:608] load_file: UNKNOWN:Failed to load file {created_time:"2023-07-27T21:32:06.97328587+00:00", filename:"/usr/share/grpc/roots.pem", children:[UNKNOWN:No such file or directory {created_time:"2023-07-27T21:32:06.973276213+00:00", errno:2, os_error:"No such file or directory", syscall:"fopen"}]}
> 
> E0727 14:32:06.974000000 22488 ssl_security_connector.cc:430] Could not get default pem root certs…..
> ```
> 
> There are steps on to do this in the README:
> 
> https://github.com/InfluxCommunity/influxdb3-python/blob/70ceac05026a143cb8af953c306111149f89c06a/README.md?plain=1#L166-L200

Hi Joshua, I have tried your code and got an issue for the flight_client_options. Would you please help to look at it? Thanks.

client = InfluxDBClient3(
      8     host=host, 
      9     token=token, 
     10     org=org,
     11     database="Pokemon",
---> 12     flight_client_options = flight_client_options(
     13         tls_root_certs=cert)
     14 )
     17  # client = InfluxDBClient3.InfluxDBClient3( 
     18  #     token="", 
     19  #     host="b0c7cce5-8dbc-428e-98c6-7f996fb96467.a.influxdb.io", 
   (...)
     22  #     flight_client_options=flight_client_options( 
     23  #         tls_root_certs=cert)) 

TypeError: 'dict' object is not callable

**Comment by powersj** [comment]
@FangqiZhu,

Looks like you are using a notebook of some sort? I'm wondering if you have some sort of syntax issue or did not import the flight_client_options? What version of the library are you using?

```python3
python3
Python 3.12.4 (main, Jun  7 2024, 06:33:07) [GCC 14.1.1 20240522] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import influxdb_client_3 as InfluxDBClient3
>>> from influxdb_client_3 import flight_client_options
>>> import certifi
>>> fh = open(certifi.where(), "r")
>>> cert = fh.read()
>>> fh.close()
>>> client = InfluxDBClient3.InfluxDBClient3(
...      token="",
...      host="b0c7cce5-8dbc-428e-98c6-7f996fb96467.a.influxdb.io",
...      org="6a841c0c08328fb1",
...      database="flightdemo",
...      flight_client_options=flight_client_options(
...          tls_root_certs=cert))
>>> table = client.query(
...      query="SELECT * FROM flight WHERE time > now() - 4h",
...      language="influxql")
```

**Comment by FangqiZhu** [comment]
Yes, I am using the jupyter notebook on a window laptop. I have just copied and pasted your example code to have a try and it showed that problem. I am using the latest version of the influxdb_client_3 (I just pip install it). Do you have any ideas how to modify the flight_client_options? Thanks.

**Comment by FangqiZhu** [comment]
Yeah, you are right. I am using the jupyter notebook on a windows laptop. Previously I was using "from influxdb_client_3 import InfluxDBClient3" instead of "import influxdb_client_3 as InfluxDBClient3" such that the influxdbClient3 cannot call the dot method InfluxDBClient3.InfluxDBClient3,  (https://us-east-1-1.aws.cloud2.influxdata.com/orgs/afa3411cee6b001e/new-user-setup/python). If I directly use client = InfluxDBClient3(host=host, token=token, org=org, flight_client_options), it seems like it cannot run successfully. Now I figure it out. Thanks for the hint.

**Comment by bednar** [comment]
The issue was caused by using the `influxdb_client_3` with an InfluxDB 2 account. This client is specifically designed to support only InfluxDB 3.

**Comment by shanmukh-d** [comment]
> Gotcha. The tutorial on the website (Load Data -> Clients -> Python) directs me to use the v3 python client. Maybe v2 users could be directed to a different tutorial? As a new user, it took me while to figure out which version I was using and which client to use.

Even i got confused by this , how to figure out which version of influx am i using ? my host is https://us-east-1-1.aws.cloud2.influxdata.com/ does that mean i am using v2 ?

**Comment by Tgi0** [comment]
The corrective steps from README used to work for me until a couple weeks ago and now they do not, I have the same messages 
`
E1106 16:02:20.075000000 92744 ssl_utils.cc:606] load_file: UNKNOWN:Failed to load file {created_time:"2024-11-06T15:02:19.9886988+00:00", filename:"/usr/share/grpc/roots.pem", children:[UNKNOWN:No such file or directory {created_time:"2024-11-06T15:02:19.988693+00:00", errno:2, os_error:"No such file or directory", syscall:"fopen"}]}
E1106 16:02:20.083000000 92744 ssl_security_connector.cc:425] Could not get default pem root certs.
E1106 16:02:20.085000000 92744 chttp2_connector.cc:263] Failed to create channel args during subchannel creation: INTERNAL: Failed to create secure subchannel for secure name 'eu-central-1-1.aws.cloud2.influxdata.com:443'; Got args: {grpc.client_channel_factory=0x26a205a0b60, grpc.default_authority=eu-central-1-1.aws.cloud2.influxdata.com:443, grpc.initial_reconnect_backoff_ms=100, grpc.internal.channel_credentials=0x26a20925b40, grpc.internal.event_engine=0x26a3b091820, grpc.internal.subchannel_pool=0x26a21a52920, grpc.max_receive_message_length=-1, grpc.primary_user_agent=grpc-c++/1.51.1, grpc.resource_quota=0x26a21a4c650, grpc.secondary_user_agent=influxdb3-python/0.8.0, grpc.server_uri=dns:///eu-central-1-1.aws.cloud2.influxdata.com:443, grpc.use_local_subchannel_pool=1}
An unexpected error occurred: Flight returned unavailable error, with message: empty address list: . gRPC client debug context: UNKNOWN:empty address list:  {grpc_status:14, created_time:"2024-11-06T15:02:20.1073024+00:00"}. Client context: IOError: Server never sent a data message. Detail: Internal`

**Comment by bednar** [comment]
Hi @Tgi0, is this issue reproducible? How do you manage your SSL certificates? Are you using certifi?

**Comment by Tgi0** [comment]
Hi, thanks for the reply. 
I have no idea about certificate management. I have contacted my IT support for help. All I can say is that the following fix
```
import certifi 
  
 fh = open(certifi.where(), "r") 
 cert = fh.read() 
 fh.close() 
```
Used to work but doe snot any more although it still seems to get the certs normally. I tried updating package certifi. I have not changed anything myself, I don't know if this suggests the problem is purely on my side.
